tests/common/mod.rs

```diff
@@ -580,6 +608,43 @@ pub(crate) fn do_channel_full_cycle<E: ElectrumApi>(
 	assert_eq!(node_b.payment(&payment_id).unwrap().amount_msat, Some(determined_amount_msat));
 	assert!(matches!(node_b.payment(&payment_id).unwrap().kind, PaymentKind::Bolt11 { .. }));
 
+	// Test manually registered/claimed payments.
+	let invoice_amount_3_msat = 5532_000;
+	let manual_preimage = PaymentPreimage([42u8; 32]);
+	let manual_payment_hash = PaymentHash(Sha256::hash(&manual_preimage.0).to_byte_array());
+	let manual_invoice = node_b
+		.bolt11_payment()
+		.receive_for_hash(invoice_amount_3_msat, &"asdf", 9217, manual_payment_hash)
+		.unwrap();
+	let manual_payment_id = node_a.bolt11_payment().send(&manual_invoice).unwrap();
+
+	let claimable_amount_msat = expect_payment_claimable_event!(
+		node_b,
+		manual_payment_id,
+		manual_payment_hash,
+		invoice_amount_3_msat
+	);
+	node_b
+		.bolt11_payment()
+		.claim_for_hash(manual_payment_hash, claimable_amount_msat, manual_preimage)
```

Comment:
Is it worth testing another payment and calling `fail_for_hash()`?

Reply:
Fair enough. Sounds good!

Reply:
It would test that the storage status is updated from `Pending` to `Failed`, though.